File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>m;
        for(int i=0;i<nums.size();i++)
        {
            int curr = nums[i];
            int rem = target - curr;
            if(m.find(rem)!=m.end())
            {
                return {i,m[rem]};
            }
            m[curr] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int res[] = new int[2];
        
        HashMap<Integer,Integer> m = new HashMap<>();
        
        for(int i=0;i<nums.length;i++){
            
            int rem = target - nums[i];
            
            if(m.containsKey(rem) == true){
                res[0] = i;
                res[1] = m.get(rem);
            }
            else{
                m.put(nums[i],i);
            }
            
        }
        
        
        
        
        return res;
    }
}